Readme.md
= Hyphenation for TYPO3 =

This is the TYPO3 extension "nkhyphenation". Its manual is available at

http://docs.typo3.org/typo3cms/extensions/


Changelog.txt

v1.0.4
2014-01-19 [TASK] Increased version to 1.0.4, added changelog. (Commit abf8ab492d3c8e42503846506ed87b5d7d069157)
2014-01-19 [BUGFIX] Made unit tests work again with TYPO3 6.2. (Commit c015c06fbb9d5a6cf62c711448df7de451d98a11)
2014-01-19 [BUGFIX] Fixed encoding when PHP default_charset is not UTF-8. (Commit e12313a62283325b5be855d7b26a1448e3b3a8e1)
2014-01-16 [TASK] Added copyright headers to most files. (Commit 8fc537739c60e152e626533482efc5cea31b4dbf)
2014-01-05 [BUGFIX] Use "jQuery" instead of "$" in C&P-Script. (Commit c16c943d2d4683c4152f20d559476041e8c1d106)

v1.0.3
2014-01-05 [TAKS] Increased version to 1.0.3. (Commit 0021b1fd87b005d9a957071cc3c9c7c643a926f7)
2014-01-05 [TASK] Added static template for CSC 6.2. (Commit 4d8860e9c988dd974c99b88939637d8cd01cfe7f)
2014-01-05 [BUGFIX] Respect HTML tags again when hyphenating CSC. (Commit e04ac5001293ff1c0997059858a6a3fb736040f6)
2014-01-05 [TASK] Added a test. (Commit 54f53d75e4b0c10928342d80dc73d84bcdb234e7)
2014-01-05 [BUGFIX] Use ObjectManager to create objects. (Commit bcfa4405e5caf24c3daacc46a1f3dceea570b90c)
2014-01-05 [TASK] Allow TYPO3 6.2 in requirements. (Commit dd45128f56acbebb3619a13234c051370f318e56)

v1.0.2
2014-01-04 [TASK] Increase version to 1.0.2. (Commit 61c95f822ef7904958947c4142f27218dee0184b)

v1.0.1
2014-01-04 [TASK] Increased version to 1.0.1 for release. (Commit 25df823bdf853b3485ac55460fa474109cbd562d)
2014-01-04 [BUGFIX] Update version requirements. (Commit df59166940dfa0d6f9f5315618340d8682c71930)
2014-01-04 [TASK][BUGFIX] Documentation improvement and fixes. (Commit c6e7ffb04ad22e248c054aed6b14719d61fba6ea)
2014-01-04 [BUGFIX] Fix hyphenate.preserveHtmlTags in stdWrap. (Commit 078ff3e89a81e7e8bbb65acb1d41a4654f687a05)
2014-01-04 [TASK] Conformity to metrics.typo3.org (Commit 4b70fe65566dbd82f6223cd9088bc04aeca9ee0a)

v1.0.0
2013-12-01 Set state to "beta". (Commit 8a1476a7801473d35ebef5dd7eb23266e1c0e193)
2013-12-01 Added Readme.md. (Commit 7df0f26af6fadce617187189ba21337823017ad4)
2013-12-01 Added information about the JS-script to the manual. (Commit 19b958c24ce16d84a3b6c96eada90c508c03b292)
2013-12-01 Added check to see if a pattern set was actually found. If not, simply return the non-hyphenated content. (Commit 46b404a578b493c1c023d066415d410bf21909ad)
2013-12-01 Removed debug code. (Commit a03a786226db1cf648532137d7b6201d4fdcff67)
2013-12-01 Added static template for CSC 6.0. (Commit a6fe701f6149057f79e4f9e83b9ef9b1d11c3456)
2013-12-01 Implemented interface of hyphenate-ViewHelper. (Commit d420338565cf035edef41d3acd409e0d386d4358)
2013-12-01 Added first version of a manual. (Commit 6e1471786580fd69c0d092ac9247fa2a2a2bde9a)
2013-12-01 Changed maximum supported PHP version to a realistic value. (Commit 71e3f4b542f1af0a99a636069835b6916c00b4b0)
2013-12-01 Changed default behaviour of hyphenate.preserveHtmlTags to "preserve". (Commit ccf3adcc28ed97db6948f8377fad542825424027)
2013-12-01 Added JavaScript to remove hyphens when copying. (Commit 6e200404c6527677270015c0d8cfb403863c48f8)
2013-12-01 Made splitting of text into words case insensitive. (Commit 864d1aef7c5f0a81eae64bf841d10bf690511c8b)
2013-11-25 Added static template to enable hyphenation in tt_content. (Commit 4e03751f1f9021d8bde528aff7b45e552c46b66f)
2013-11-25 Fixed constraints-description in ext_emconf.php (Commit 3ee469e1a5ee55d3f8c1ac6200811558a23a649a)
2013-11-25 Updated csh-descriptions for hyphenation patterns. (Commit d0f41c4269a98d0e288e8362443cfd59d2ab7c42)
2013-11-25 Made "hidden" property appear in backend forms. (Commit b591ae923e6393a42a72bb2e1a4aac0f6e0a503d)
2013-11-25 Switched to a more thought-through way handling leftmin, rightmin and wordcharacters from files. (Commit f55883bf3a0376ae2765295bfc71bc4b4ee868bc)
2013-11-24 Removed some commented-out code from hyphenation patterns test. (Commit 9a11ca53bd46147e2aee19c965e1f95036ff2ee6)
2013-11-24 Added hook to stdWrap for hyphenation. (Commit 6b039c694c10f9b4921a5370c5b57aa38414ed5a)
2013-11-24 Added default language to TCA for system language setting. (Commit e39b7359b0b5f448ac5e1c9aec14ecc185d256cd)
2013-11-24 Decode entities in the hyphenation pattern to get consistent results when hyphenating HTML fragments and normal text (DOMDocument does not print entities) (Commit cbabd4778afff11a7078e93cbfae29dbbc8537c8)
2013-11-24 Removed unused hook file. (Commit 96b8f51dc22876c18c8bcfa1f0a33ba598025dbf)
2013-11-23 Added the possibility to keep HTML tags. (Commit b3672a848b43e9d14c833f084f9029b66fa6ec3c)
2013-11-23 Made unit tests rund green again and added a test. (Commit 5fd08170eb19fac8b1bbd5e0b617273475ad827c)
2013-11-23 Implemented hyphenation viewhelper, use FAL for file handling and the caching framework for trie-storage. (Commit 66b94d8ac65955e4e49ce97cdd58f4145dfb4e27)
2013-11-17 Moved hook class to namespaces. (Commit 4042fea6bc0dbcd9a501e1bb97a3950f3401279a)
2013-11-17 Removed unused classes and files. (Commit 5329f72a63820189008885b3f1210bf7ed5ca245)
2013-11-17 Removed backend module registration. (Commit 073dbce7063dcff2fe33dfb70a69c2eca5e55c86)
2013-11-17 Uncommented assertion. (Commit 23b391bd7b3cf6ccefb8ff888a4d737693a38b60)
2013-11-17 Added HyphenatorJSPatternProvider, which is able to parse Hyphenator.js pattern files. (Commit 9c0000e70188eb476972a5f5d594e4b93f734682)
2013-11-17 Changed word start/end marker for patterns and added tests for it. (Commit 0024311c9f65e7b0cb5588cc0243ccbcd1e1757b)
2013-11-17 Moved hyphenation algorithm to hyphenation pattern domain model. (Commit d90945c6a24651884e8c765ecd18d6578f2b25e1)
2013-11-17 Renamed "special characters" to "word characters" in all remaining places. (Commit dc5834974632a3cf770ecfdfc5b5b2bd77c4860c)
2013-11-16 Added some more tests for hyphenation algorithm to check if it can cope with multiple splitting points. (Commit bb3b53d501246aae16f2e33ad214bc83b6880cf3)
2013-11-16 Replaced another occurence of str_split with preg_split. (Commit a5935fc3fb80e4605e09aab4c00c3077645588c8)

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "nkhyphenation".
 *
 * Auto generated 04-01-2014 14:34
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
	'title' => 'Server side hyphenation for TYPO3',
	'description' => 'Adds soft hyphenation capabilities to TYPO3. Provides a ViewHelper, a stdWrap property and some static templates to enable hyphenation.',
	'category' => 'fe',
	'shy' => 0,
	'version' => '1.0.4',
	'priority' => '',
	'loadOrder' => '',
	'module' => '',
	'state' => 'beta',
	'uploadfolder' => 1,
	'createDirs' => 'uploads/tx_nkhyphenation/',
	'modify_tables' => '',
	'clearcacheonload' => 1,
	'lockType' => '',
	'author' => 'Jost Baron',
	'author_email' => 'j.baron@netzkoenig.de',
	'author_company' => 'Netzkönig GbR',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => 
	array (
		'conflicts' => 
		array (
		),
		'suggests' => 
		array (
		),
		'depends' => 
		array (
			'typo3' => '6.0.0-6.2.99',
			'php' => '5.3.7-5.5.99',
			'fluid' => '6.0.0-6.2.99',
			'extbase' => '6.0.0-6.2.99',
		),
	),
	'_md5_values_when_last_written' => 'a:43:{s:16:"ext_autoload.php";s:4:"ad3f";s:12:"ext_icon.gif";s:4:"e922";s:17:"ext_localconf.php";s:4:"296b";s:14:"ext_tables.php";s:4:"9f44";s:14:"ext_tables.sql";s:4:"8c7a";s:44:"Classes/Domain/Model/HyphenationPatterns.php";s:4:"a85f";s:59:"Classes/Domain/Repository/HyphenationPatternsRepository.php";s:4:"7e8e";s:27:"Classes/Hooks/BuildTrie.php";s:4:"4d1d";s:38:"Classes/Service/HyphenationService.php";s:4:"9cc6";s:36:"Classes/Utility/BuildTrieUtility.php";s:4:"1c12";s:41:"Configuration/Tca/hyphenationpatterns.php";s:4:"ec9c";s:40:"Resources/Private/Language/locallang.xlf";s:4:"1a15";s:40:"Resources/Private/Language/locallang.xml";s:4:"231f";s:43:"Resources/Private/Language/locallang_db.xlf";s:4:"5c75";s:43:"Resources/Private/Language/locallang_db.xml";s:4:"c278";s:64:"Resources/Private/Language/locallang_hyphenationpatterns_csh.xlf";s:4:"25c3";s:64:"Resources/Private/Language/locallang_hyphenationpatterns_csh.xml";s:4:"05ac";s:33:"Resources/Private/patterns/bn.php";s:4:"32f7";s:33:"Resources/Private/patterns/de.php";s:4:"9fea";s:33:"Resources/Private/patterns/en.php";s:4:"81b5";s:33:"Resources/Private/patterns/es.php";s:4:"6636";s:33:"Resources/Private/patterns/fi.php";s:4:"c910";s:33:"Resources/Private/patterns/fr.php";s:4:"027f";s:33:"Resources/Private/patterns/gu.php";s:4:"96d2";s:33:"Resources/Private/patterns/hi.php";s:4:"cfba";s:33:"Resources/Private/patterns/it.php";s:4:"56e7";s:33:"Resources/Private/patterns/ka.php";s:4:"f609";s:33:"Resources/Private/patterns/ml.php";s:4:"9702";s:33:"Resources/Private/patterns/nl.php";s:4:"f055";s:33:"Resources/Private/patterns/or.php";s:4:"c016";s:33:"Resources/Private/patterns/pa.php";s:4:"0592";s:33:"Resources/Private/patterns/pl.php";s:4:"fa1f";s:33:"Resources/Private/patterns/ru.php";s:4:"2285";s:33:"Resources/Private/patterns/sv.php";s:4:"cab7";s:33:"Resources/Private/patterns/ta.php";s:4:"3446";s:33:"Resources/Private/patterns/te.php";s:4:"bac9";s:51:"Tests/Unit/Domain/Model/HyphenationPatternsTest.php";s:4:"17c9";s:45:"Tests/Unit/Service/HyphenationServiceTest.php";s:4:"9d9e";s:28:"nbproject/project.properties";s:4:"7a11";s:21:"nbproject/project.xml";s:4:"6c0d";s:35:"nbproject/private/config.properties";s:4:"d41d";s:36:"nbproject/private/private.properties";s:4:"aab6";s:29:"nbproject/private/private.xml";s:4:"0b1a";}',
);

?>